cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -51,48 +70,63 @@ class MemoryMappedDirectedGraph(file: File) extends DirectedGraph[Node] {
   private class MemoryMappedDirectedNode(override val id: Int) extends Node {
     val nodeOutboundOffset = outboundOffset(id)
     val nodeInboundOffset = inboundOffset(id)
-    override def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = outDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeOutboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
-    override def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = inDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeInboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
   }
 
-  override def getNodeById(id: Int): Option[Node] =
+  def getNodeById(id: Int): Option[Node] =
     if(0 <= id && id < nodeCount) {
       Some(new MemoryMappedDirectedNode(id))
     } else {
       None
     }
 
-  override def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
+  def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
 
-  override def edgeCount: Long = (outboundOffset(nodeCount) - outboundOffset(0)) / 4
+  lazy val edgeCount: Long = outboundOffset(nodeCount) - outboundOffset(0)
```

Comment:
Why declare this as a lazy val as opposed to a function? I can imagine a version of this class where 'data' is a var as opposed to a val -- so one could potentially just override data and everything might keep working.